command/operator_diagnose.go

```diff
@@ -140,6 +152,61 @@ func (c *OperatorDiagnoseCommand) RunWithParsedFlags() int {
 		return 1
 	}
 
+	// Check Listener Information
+	// TODO: Run Diagnose checks on the actual net.Listeners
+
+	disableClustering := config.HAStorage.DisableClustering
+	infoKeys := make([]string, 0, 10)
+	info := make(map[string]string)
+	status, lns, _, errMsg := server.InitListeners(config, disableClustering, &infoKeys, &info)
+
+	if status != 0 {
+		c.UI.Output("Error parsing listener configuration.")
+		c.UI.Error(errMsg.Error())
+		return 1
+	}
+
+	// Make sure we close all listeners from this point on
+	listenerCloseFunc := func() {
+		for _, ln := range lns {
+			ln.Listener.Close()
+		}
+	}
+
+	defer c.cleanupGuard.Do(listenerCloseFunc)
+
+	sanitizedListeners := make([]listenerutil.Listener, 0, len(config.Listeners))
+	for _, ln := range lns {
+		if ln.Config.TLSDisable {
+			c.UI.Warn("WARNING! TLS is disabled in a Listener config stanza.")
+			continue
+		}
+		if ln.Config.TLSDisableClientCerts {
+			c.UI.Warn("WARNING! TLS for a listener is turned on without requiring client certs.")
+		}
+
+		// Check ciphersuite and load ca/cert/key files
+		// TODO: TLSConfig returns a reloadFunc and a TLSConfig. We can use this to
+		// perform an active probe.
+		_, _, err := listenerutil.TLSConfig(ln.Config, make(map[string]string), c.UI)
+		if err != nil {
+			c.UI.Output("Error creating TLS Configuration out of config file: ")
+			c.UI.Output(err.Error())
+			return 1
+		}
+
+		sanitizedListeners = append(sanitizedListeners, listenerutil.Listener{
+			Listener: ln.Listener,
+			Config:   ln.Config,
+		})
+	}
+	err = diagnose.ListenerChecks(sanitizedListeners)
+	if err != nil {
+		c.UI.Output("Diagnose caught configuration errors: ")
```

Comment:
I'm not familiar with `c.UI.Output` - is the trailing space intended, or will that be auto added? Basically, is this making us have two spaces between "Diagnose..." and the error?

Reply:
The trailing space is intended -- from my testing, it isn't added automatically.